src/reg_allocator.hpp

```diff
@@ -0,0 +1,80 @@
+/*
+This is a part of Loops project.
+Distributed under Apache 2 license.
+See https://github.com/vpisarev/loops/LICENSE
+*/
+
+#ifndef __LOOPS_REG_ALLOCATOR_HPP__
```

Comment:
overall, the code looks nice.

But, there should be some explanation in the beginning, some expanded comment what the algorithm does and what's the theoretical complexity, what's the space overhead. Those are all quite important things. In general, we want our compiler to have O(N) time complexity and O(N) space overhead, where N is the number of instructions. Register allocator is one of the most complex parts in this compiler (so far), so it's important to provide some good description and explain why it's O(N). (and fix if it's not)

Reply:
Done.